tests/metrics/test_superset.py

```diff
@@ -0,0 +1,291 @@
+    result = super_set_fairness_metrics(
+        metric,
+        data,
+        sensitive,
+        real_target,
+        predicted_target,
+    )
+    assert result == expected_result
+
+
+super_set_performance_metrics_expected_result_2 = [
+    {
+        "sensitive": ("Sex",),
+        "result": (
+            ["HeartDisease"],
+            [
+                {
+                    "sensitive": np.array(["M"], dtype=object),
+                    "selection_rate_in_data": np.array(0.63172414),
+                    "selection_rate_in_predictions": np.array(0.6262069),
+                    "accuracy": [0.9779310344827586],
+                    "balanced accuracy": [np.float64(0.9778470143761346)],
+                    "precision": [np.float64(0.986784140969163)],
+                    "recall": [np.float64(0.9781659388646288)],
+                    "f1-score": [np.float64(0.9824561403508771)],
+                    "false_negative_rate": [np.float64(0.021834061135371178)],
+                    "false_positive_rate": [np.float64(0.02247191011235955)],
+                    "true_negative_rate": [np.float64(0.9775280898876404)],
+                    "true_positive_rate": [np.float64(0.9781659388646288)],
+                },
+                {
+                    "sensitive": np.array(["F"], dtype=object),
+                    "selection_rate_in_data": np.array(0.25906736),
+                    "selection_rate_in_predictions": np.array(0.2642487),
+                    "accuracy": [0.9637305699481865],
+                    "balanced accuracy": [np.float64(0.956013986013986)],
+                    "precision": [np.float64(0.9215686274509803)],
+                    "recall": [np.float64(0.94)],
+                    "f1-score": [np.float64(0.9306930693069307)],
+                    "false_negative_rate": [np.float64(0.06)],
+                    "false_positive_rate": [np.float64(0.027972027972027972)],
+                    "true_negative_rate": [np.float64(0.972027972027972)],
+                    "true_positive_rate": [np.float64(0.94)],
+                },
+            ],
+        ),
+    }
+]
+
+
+@pytest.mark.parametrize(
+    "data, sensitive, real_target, predicted_target, expected_results",
+    [
+        (
+            dataset1,
+            None,
+            None,
+            None,
+            pytest.raises(ValueError),
+        ),
+        (
+            dataset2,
+            None,
+            None,
+            None,
+            super_set_performance_metrics_expected_result_2,
+        ),
+        (
+            df,
+            ["Sex"],
+            ["HeartDisease"],
+            ["HeartDiseasePred"],
+            super_set_performance_metrics_expected_result_2,
+        ),
+    ],
+)
+def test_super_set_performance_metrics(
+    data: pd.DataFrame | Dataset,
+    sensitive: Sequence[str] | None,
+    real_target: Sequence[str] | None,
+    predicted_target: Sequence[str] | None,
+    expected_results: list[dict] | RaisesContext,
+):
```

Comment:
Maybe having an if: else: that takes up the whole function is not a good idea, you could create a test just to chek the failures.

Reply:
I think I prefer a separate test, but for now the change is not necessary